vcs/repository.go

```diff
@@ -120,6 +120,8 @@ type CommitsOptions struct {
 
 	N    uint // limit the number of returned commits to this many (0 means no limit)
 	Skip uint // skip this many commits at the beginning
+
+	Path string // only commits modifying the given path are selected (optional)
```

Comment:
Yes. Is it clear or should I mention it in comment?

The upstream git docs are more detailed... or so I thought. I tried to look it up again now and I cannot find it saying it can be a file or folder anywhere. It just says this:

> Show only commits that are enough to explain how the files that match the specified paths came to be.

> <paths>
> Commits modifying the given <paths> are selected.

Note that it can be multiple paths (in the git command, but the API change I propose in this PR only allows setting one path max).

The only relevant note I could find is docs for --follow flag:

> --follow
> Continue listing the history of a file beyond renames (works only for a single file).

But I think I verified it works on paths that folders through empirical testing.